packages/blog-starter-kit/themes/hashnode/components/toc-render-design.tsx

```diff
@@ -7,226 +7,235 @@ import { ChevronDownSVG_16x16, ChevronRightSVG_16x16, ChevronUpSVG_16x16 } from
 }
 
 interface TocTreeProps {
-  list: any[];
-  minHeaderLevel?: number;
-  currentItem?: any;
-  numItemsCompleted?: number;
-  modal?: boolean;
+	list: any[];
+	minHeaderLevel?: number;
+	currentItem?: any;
+	numItemsCompleted?: number;
+	modal?: boolean;
 }
 
 interface TocRenderDesignProps {
-  list: any[];
-  hideShowMoreOption?: boolean;
-  modal?: boolean;
+	list: any[];
+	hideShowMoreOption?: boolean;
+	modal?: boolean;
 }
 
 function TocRow(props: TocRowProps) {
-  const { children, node, modal } = props;
-  const [childrenVisibility, setChildrenVisibility] = useState(false);
-  const { hide: hideTocModal } = useTocModalStore();
-
-  return (
-    <li key={node.id} className="px-2 py-0.5 align-middle">
-      <div
-        className={twJoin(
-          'flex items-center gap-2 rounded-lg',
-          modal ? 'hover:bg-slate-100 dark:hover:bg-slate-800' : 'hover:underline',
-        )}
-      >
-        {node.hasChildren && (
-          <button
-            type="button"
-            className="pl-2.5 text-slate-400"
-            onClick={() => {
-              setChildrenVisibility((prevVisibility: boolean) => !prevVisibility);
-            }}
-          >
-            {childrenVisibility ? (
-              <ChevronDownSVG_16x16 className="h-4 w-4 stroke-current" />
-            ) : (
-              <ChevronRightSVG_16x16 className="h-4 w-4 stroke-current" />
-            )}
-          </button>
-        )}
-        <a
-          id={node.id}
-          className={twJoin(
-            'w-full py-2.5 pr-2.5 text-sm font-medium text-slate-800 dark:text-slate-100',
-            !node.hasChildren && 'pl-3',
-          )}
-          href={`#heading-${node.slug}`}
-          onClick={() => {
-            if (hideTocModal && modal) {
-              // This rAF is required to prevent flickering of the navbar when the user clicks on a link in the TOC modal
-              requestAnimationFrame(() => {
-                hideTocModal();
-              });
-            }
-          }}
-          // eslint-disable-next-line react/no-danger
-          dangerouslySetInnerHTML={{
-            __html: node.title,
-          }}
-          aria-label={node.title}
-        />
-      </div>
-      {node.hasChildren ? <> {childrenVisibility && <div className="pl-5">{children}</div>} </> : <>{children}</>}
-    </li>
-  );
+	const { children, node, modal } = props;
+	const [childrenVisibility, setChildrenVisibility] = useState(false);
+	const { hide: hideTocModal } = useTocModalStore();
+
+	return (
+		<li key={node.id} className="px-2 py-0.5 align-middle">
+			<div
+				className={twJoin(
+					'flex items-center gap-2 rounded-lg',
+					modal ? 'hover:bg-slate-100 dark:hover:bg-slate-800' : 'hover:underline',
+				)}
+			>
+				{node.hasChildren && (
+					<button
+						type="button"
+						className="pl-2.5 text-slate-400"
+						aria-label={childrenVisibility ? `Collapse ${node.title}` : `Expand ${node.title}`}
+						aria-expanded={childrenVisibility}
```

Comment:
`childrenVisibility` controls the visibility of child nodes in the TOC.
It toggles between true (expanded) and false (collapsed) when the user clicks the button to expand or collapse a TOC section.